cassiopeia/data/CassiopeiaTree.py

```diff
@@ -897,7 +896,6 @@ def compute_dissimilarity_map(
             weights,
             self.missing_state_indicator,
         )
-        dissimilarity_map = scipy.spatial.distance.squareform(dissimilarity_map)
```

Comment:
I actually don't know if we should be doing this. Part of me thinks, actually, that we should be working with flattened dissimilarity matrices because they can get so large! (Think about a lineage with 1000 cells -- the square matrix will have 1000^2 = 1e6 entries when in reality half of those are duplicated).

Reply:
Do you mean we should change all of our functions to work with flattened dissimilarity maps? Or do you mean that there's a way to pass a flattened array into a pandas dataframe. I think the issue is that with flattened arrays it would be difficult to maintain indices (as we do in the next line when we transform the dissimilarity map into a pandas dataframe). I think that if we do not transform it into a df it might hurt usability

Reply:
Ideally, we should change all of our solver methods to work with flattened dissimilarity matrices. It's just a lot more efficient! But that's not necessarily a top priority right now. Let's just not return, by default, a full dissimilarity matrix -- I'd rather have to convert it to squareform elsewhere.